### PR TITLE
[full-ci][tests-only] Improve then steps for suite 'apiFederationToShares2'

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Improve then steps for acceptance tests suite `apiFederationToShares2`

## Related Issue
- Part of https://github.com/owncloud/core/issues/39512

### CODE REVIEW

**Comment by phil-davis** [comment]
@kiranparajuli589 needs a rebase:
Conflicting files
tests/acceptance/features/bootstrap/WebDav.php 

A few other testing PRs have been merged today.

**Comment by kiranparajuli589** [comment]
> @kiranparajuli589 needs a rebase: Conflicting files tests/acceptance/features/bootstrap/WebDav.php
> 
> A few other testing PRs have been merged today.

:100: fixed it PhilJI

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>apiSharingNotificationsToShares-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/35401/102

**Review by SagarGi — COMMENTED** [review]
we can put a comment for the OCS status code where `notset` is cheked as it is asserted so that it could be more clear what `notset` is for others. Others LGTM.

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -3591,7 +3603,7 @@ public function userHasCreatedFolder(string $user, string $destination):void {
 			["201", "204"],
 			"HTTP status code was not 201 or 204 while trying to create folder '$destination' for user '$user'"
 		);
-		$this->emptyLastHTTPStatusCodesArray();
+		$this->clearStatusCodeArrays();
```
Comment (SagarGi) [inline tests/acceptance/features/bootstrap/WebDav.php:3606]: a question was raised against me for emptying an OCS status code for the steps not giving OCS status code. @amrita-shrestha what do you think?

**Review by phil-davis — APPROVED** [review]
LGTM if CI passes.

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/WebDav.php]: :+1: fixed

**Pull request merged**